core/io/resource_uid.cpp

```diff
@@ -311,7 +311,7 @@ String ResourceUID::get_path_from_cache(Ref<FileAccess> &p_cache_file, const Str
 		ERR_FAIL_COND_V(rl != len, String());
 
 		if (singleton->id_to_text(id) == p_uid_string) {
-			return String(cs.get_data());
+			return String::utf8(cs.get_data());
```

Comment:
You could make use of `Span` here:
```c++
		ERR_FAIL_COND_V(cs.resize(len) != OK, String());
		int32_t rl = p_cache_file->get_buffer((uint8_t *)cs.ptrw(), len);
 		ERR_FAIL_COND_V(rl != len, String());

 		if (singleton->id_to_text(id) == p_uid_string) {
 			return String::utf8(cs);
 		}
```
This would notably stop using `strlen`, so it would stop cutting the string early if `\0` was found.